test/floki/html_tree_test.exs

```diff
@@ -256,4 +256,42 @@ defmodule Floki.HTMLTreeTest do
     refute Enum.member?(html_tree, %{html_node3 | type: "marquee"})
     refute Enum.member?(html_tree, 42)
   end
+
+  test "Inspect works with HTMLTree" do
+    html_tree = %HTMLTree{
+      root_nodes_ids: [2, 1],
+      node_ids: [2, 1],
+      nodes: %{
+        1 => %Text{content: "hello", node_id: 1},
+        2 => %Text{content: " world", node_id: 2}
+      }
+    }
```

Comment:
After thinking for a while, I believe the parser won't produce such tree, because the `Text` would be collapsed`. So I would change this to one `Text` node, or maybe adding a tag in between.